src/history.cpp

```diff
@@ -1453,7 +1453,10 @@ bool history_t::save_internal_via_rewrite()
                 if (wstat(new_name, &sbuf) >= 0)
                 {
                    /* Success */
-                    fchown(out_fd, sbuf.st_uid, sbuf.st_gid);
+                    if (0 > fchown(out_fd, sbuf.st_uid, sbuf.st_gid))
+                    {
+                        debug(2, L"Error when changing ownership of history file");
```

Comment:
Include the errno in the message. I think this is also serious enough to warrant level one.